src/expr.rs

```diff
@@ -662,7 +662,30 @@ fn rewrite_if_else(context: &RewriteContext,
             _ => else_block.rewrite(context, width, offset),
         };
 
-        result.push_str(" else ");
+        let snippet = context.codemap
+                             .span_to_snippet(mk_sp(if_block.span.hi, else_block.span.lo))
+                             .unwrap();
+        // Preserve comments that are between the if and else block
+        if contains_comment(&snippet) {
+            let close_pos = try_opt!(snippet.find_uncommented("else"));
+            let trimmed = &snippet[..close_pos].trim();
```

Comment:
What happens if the comment is after the `else` but before the else block? (Needs a test too).